plugin.video.imdb.trailers/addon.py

```diff
@@ -0,0 +1,326 @@
+    view_mode = int(_settings('VideoMode'))
+
+CACHE_FILE = xbmc.translatePath('special://profile/addon_data/' + _addonID + '/requests_cache')
+requests_cache.install_cache(CACHE_FILE, backend='sqlite', expire_after=int(_settings('timeout'))*3600)
+
+CONTENT_URL = 'https://www.imdb.com/trailers/'
+SHOWING_URL = 'https://www.imdb.com/movies-in-theaters/'
+COMING_URL = 'https://www.imdb.com/movies-coming-soon/{}-{:02}'
+DETAILS_PAGE = "https://m.imdb.com/videoplayer/{}"
+USER_AGENT = 'Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.17 (KHTML, like Gecko) Chrome/24.0.1312.57 Safari/537.17'
+
+if not xbmcvfs.exists('special://profile/addon_data/'+_addonID+'/settings.xml'):
+    _addon.openSettings()
+
+class Main(object):
+    def __init__(self):
+        if ('action=list2' in sys.argv[2]):
+            self.list_contents2()
+        elif ('action=list' in sys.argv[2]):
+            self.list_contents()
+        elif ('action=play' in sys.argv[2]):
+            self.play()
+        elif ('action=clear' in sys.argv[2]):
+            self.clear_cache()
+        else:
+            self.main_menu()
+
+
+    def main_menu(self):
+        if DEBUG:
+            self.log('main_menu()')
+        category = [{'title':_language(30201), 'key':'showing'},
+                    {'title':_language(30202), 'key':'coming'},
+                    {'title':_language(30203), 'key':'popTab'},
+                    {'title':_language(30204), 'key':'recAddTab'},
+                    {'title':_language(30205), 'key':'tvTab'},
+                    {'title':_language(30206), 'key':'cache'}]
+        for i in category:
+            listitem = xbmcgui.ListItem(i['title'])
+            listitem.setArt({'thumb': _icon,
+                             'fanart': _fanart,
+                             'icon' : _icon})
+
+            if i['key'] == 'cache':
+                url = sys.argv[0] + '?' + urllib.parse.urlencode({'action': 'clear'})
+            elif i['key'] == 'showing' or i['key'] == 'coming':
+                url = sys.argv[0] + '?' + urllib.parse.urlencode({'action': 'list2', 'key': i['key']})
+            else:
+                url = sys.argv[0] + '?' + urllib.parse.urlencode({'action': 'list', 'key': i['key']})
+            xbmcplugin.addDirectoryItems(int(sys.argv[1]), [(url, listitem, True)])
+    
+        # Sort methods and content type...
+        xbmcplugin.addSortMethod(handle=int(sys.argv[1]), sortMethod=xbmcplugin.SORT_METHOD_NONE)
+        xbmcplugin.setContent(int(sys.argv[1]), 'addons')
+        if force_mode:
+            xbmc.executebuiltin('Container.SetViewMode({})'.format(menu_mode))
+        # End of directory...
+        xbmcplugin.endOfDirectory(int(sys.argv[1]), True)
+
+
+    def clear_cache(self):
+        """
+        Clear the cache database.
+        """
+        msg = 'Cached Data has been cleared'
+        requests_cache.get_cache().clear()
+        xbmc.executebuiltin('Notification(%s, %s, %d, %s)'%(_plugin, msg, 3000, _icon))
```

Comment:
use `xbmcgui.Dialog().notification(...)`, https://codedocs.xyz/xbmc/xbmc/group__python___dialog.html#ga27157f98dddb21b44cc6456137977aa2

Reply:
Modified as per the suggestion